modules/drivers/clickhouse/test/metabase/driver/clickhouse_data_types_test.clj

```diff
@@ -337,22 +338,145 @@
+                                         (-> (lib/starts-with (lib.metadata/field mp (mt/id :nullable_uuids :uuid1))
+                                                              "AAA")
+                                             (lib.options/update-options assoc :case-sensitive false))
+                                         [[1 #uuid "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"]]]
+
+                                        ["can filter nullable uuids with ends with case sensitive lowercase"
+                                         (-> (lib/ends-with (lib.metadata/field mp (mt/id :nullable_uuids :uuid1))
+                                                            "aaa")
+                                             (lib.options/update-options assoc :case-sensitive true))
+                                         [[1 #uuid "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"]]]
+
+                                        ["can filter nullable uuids with ends with case insensitive lowercase"
+                                         (-> (lib/ends-with (lib.metadata/field mp (mt/id :nullable_uuids :uuid1))
+                                                            "aaa")
+                                             (lib.options/update-options assoc :case-sensitive false))
+                                         [[1 #uuid "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"]]]
+
+                                        ["can filter nullable uuids with ends with case sensitive uppercase"
+                                         (-> (lib/ends-with (lib.metadata/field mp (mt/id :nullable_uuids :uuid1))
+                                                            "AAA")
+                                             (lib.options/update-options assoc :case-sensitive true))
+                                         []]
+
+                                        ["can filter nullable uuids with ends with case insensitive uppercase"
+                                         (-> (lib/ends-with (lib.metadata/field mp (mt/id :nullable_uuids :uuid1))
+                                                            "AAA")
+                                             (lib.options/update-options assoc :case-sensitive false))
+                                         [[1 #uuid "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"]]]
+
+                                        ["can filter nullable uuids with is empty"
+                                         (lib/is-empty (lib.metadata/field mp (mt/id :nullable_uuids :uuid1)))
+                                         [[3 nil]]]
+
+                                        ["can filter nullable uuids with is not empty"
+                                         (lib/not-empty (lib.metadata/field mp (mt/id :nullable_uuids :uuid1)))
+                                         [[1 #uuid "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"]
+                                          [2 #uuid "bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb"]]]]]
```

Comment:
I think in most cases we describe the test cases in a data structure so that it's easier to follow what the combinations are we are testing and then interpret that data structure.